### DISCUSSION TITLE
How to reference two types to the same model?

### DISCUSSION THREAD

**Discussion by bitsmyth** [prisma/prisma#4094] [Q&A] [answered]
```prisma
model Article {
  id     String      @id @default(uuid())
  shop   Shop        @relation(fields: [shopId], references: [id])
  shopId String
  artNr  Int         
  desc   String
  imgSrc String   
}

model Rule {
  id                  String           @id @default(uuid())
  name                String
  includedArticles    Article[]         @relation("includedArticles")
  excludedArticles    Article[]         @relation("excludedArticles")
  shop                Shop              @relation(fields: [shopId], references: [id])
  shopId              String
}
```

I want to have rules for articles, where I can define that a rule includes and excludes articles. My problem is that I want also articles to exist when they are not added to any particular rule.

Let me know if this case is an logical issue from my side, otherwise I got following error message

> Error validating model "Rule": Automatic related field generation would cause a naming conflict. Please add an explicit opposite relation field.

**Comment by ryands17** [comment, marked as answer]
Hey @NikoMontana 👋 
You would need to write it like this:

```prisma
model Article {
  id            String  @id @default(uuid())
  shopId        String
  artNr         Int
  desc          String
  imgSrc        String
  includesRule  Rule?   @relation("includedArticles", fields: [includeRuleId], references: [id])
  excludesRule  Rule?   @relation("excludedArticles", fields: [excludeRuleId], references: [id])
  includeRuleId String?
  excludeRuleId String?
}

model Rule {
  id               String    @id @default(uuid())
  name             String
  includedArticles Article[] @relation("includedArticles")
  excludedArticles Article[] @relation("excludedArticles")
  shopId           String
}

```

You can read more on how to define relations [here](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#overview). An opposite relation field is required to fetch the relation from both ends.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
